src/commands/merge-layer.js

```diff
@@ -1,4 +1,90 @@
+const merge = require("lodash").assign;
+const isArray = require("lodash").isArray;
+const isString = require("lodash").isString;
+
 module.exports = function (templates) {
-  /* actual merge logic has to be written */
-  return templates[0];
+  let finalLayer = {
+    structure: {
+      directory: [],
+      file: [],
+    },
+  };
+
+  templates.forEach((layer) => {
+    let tmp = {
+      structure: mergeDirectory(finalLayer.structure, layer.structure),
+    };
+    tmp.structure.file = finalLayer.structure.file;
+
+    tmp.structure.file = mergeFile(
+      finalLayer.structure.file,
+      layer.structure.file
+    );
+    finalLayer = tmp;
+  });
+  return finalLayer;
 };
+
+function mergeDirectory(directory1, directory2) {
```

Comment:
#nitpick: Could we please use only chars for naming vars? (directoryOne, directoryTwo)